### PR TITLE
Hello Rails React - Set up

### PR DESCRIPTION
**Pull request by Donard97**
**_I opened this PR to review my project if it is as requested! 👀 👀_**

### Project requirements that are done:

> - Create a new Rails app called 'hello-rails-react'. ☑️ 
> - Initialize your project with Git. ☑️ 
> - Set up your project with webpacker the react-rails gems as describe in the previous lesson's tutorial. ☑️ 
> - Make sure that your project has a Postgres database set up. ☑️ 
> - Create a table for storing your messages and create 5 different greetings. ☑️ 
> - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route). ☑️ 
> - Create a static view that will be the root of your app. ☑️ 
> - Create the App component with react-router and render it in your static view. ☑️ 
> - Create the Greeting component that will display a greeting. Set it up as a route in your App component. ☑️ 
> - Create a store, an action and a reducer that will connect to you API endpont to get the random greeting. ☑️ 
> - Display the random greeting in your Greeting component. ☑️ 

<hr>

_Feel free to review 👁️‍🗨️ my project and if you have any questin please ask me! 👍_

### CODE REVIEW

File: app/controllers/api/greetings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Api::GreetingsController < ApplicationController
+  def index
+    greeting = Greeting.order(Arel.sql('RANDOM()')).first
+    render json: { greeting: greeting.greeting }
+  end
+end
```
Comment (petudeveloper) [inline app/controllers/api/greetings_controller.rb:6]: When trying to run your app I ran into the following error: 
```
NoMethodError (undefined method `greeting' for nil:NilClass):
  
app/controllers/api/greetings_controller.rb:4:in `index'
```
Please correct this, check the image below for a better understanding:
Screenshot from 2022-01-18 10-47-17

**Review by petudeveloper — CHANGES_REQUESTED** [review]
Hi @Donard97, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/api/greetings_controller.rb (hunk 1 above)
Comment (Donard97) [reply app/controllers/api/greetings_controller.rb:6]: Hello @petudeveloper, thank you for reviewing my project. I had this error before submitting my project but I fixed it and when I run my app there is no error. 
Here you have a screenshot of my terminal:
image

Comment (Donard97) [reply app/controllers/api/greetings_controller.rb:6]: I found the issue. Please run rails:db seed before running the server. 👍

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @Donard97 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**